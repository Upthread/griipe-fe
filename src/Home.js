import React from "react";
import firebase from "firebase";
import { BrowserRouter as Router, Link, Route, Switch } from "react-router-dom";
import FormContainer from "./components/Forms/FormContainer.jsx";

const Home = () => {
  return (
    <div>
      <button onClick={() => firebase.auth().signOut()}> Sign Out </button>
      <Link exact to="/form" component={FormContainer}>
        {" "}
        Complain{" "}
      </Link>
      <Switch>
        <Route exact path="/" component={Home} />

        <Route exact to="/form" component={FormContainer} />
        <Route
          path="/blog"
          children={({ match }) => (
            <li className={match ? "active" : ""}>
              <Link to="/blog">Blog</Link>
            </li>
          )}
        />
        <Route render={() => <h1>Page not found</h1>} />
      </Switch>
    </div>
  );
};
export default Home;
