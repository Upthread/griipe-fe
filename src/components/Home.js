import React, {Component} from 'react'
import {BrowserRouter as Router, Route, Link} from 'react-router-dom';
import firebase from 'firebase'

import Navigation from './Navigation.js';
import ComplaintCard from './Feeds/ComplaintCard.js';
import RecordForm from './Forms/RecordForm.js'

import MaterialIcon, {colorPalette} from 'material-icons-react';



export default class SignedIn extends Component {
    state = {
        "cards": [
           {
                "id": 1,
                "name": "Tim",
                "uid": "00mKY5N1WBRO7mGcfpvFDzSGkJJ3",
                "complaint": "Lorem ipsum dolor amet copper mug labore neutra, artisan salvia squid green juice williamsburg literally ea typewriter small batch humblebrag seitan. Trust fund proident enim, activated charcoal pop-up chambray shabby chic officia woke. Church-key austin next level, scenester dreamcatcher succulents asymmetrical drinking vinegar. Keytar reprehenderit in hammock woke eu gochujang incididunt before they sold out ut chartreuse excepteur man bun locavore. Vape activated charcoal quinoa hoodie try-hard artisan YOLO nisi dolor cardigan ugh. Pork belly salvia iPhone gluten-free retro you probably haven't heard of them seitan scenester whatever est chartreuse semiotics brooklyn hexagon gastropub."
           },
           {
                "id": 2,
                "name": "Tom",
                "uid": "00mKY5N1WBRO7mGcfpvFDzSGkJJ3",
                "complaint": "Lorem ipsum dolor amet copper mug labore neutra, artisan salvia squid green juice williamsburg literally ea typewriter small batch humblebrag seitan. Trust fund proident enim, activated charcoal pop-up chambray shabby chic officia woke. Church-key austin next level, scenester dreamcatcher succulents asymmetrical drinking vinegar. Keytar reprehenderit in hammock woke eu gochujang incididunt before they sold out ut chartreuse excepteur man bun locavore. Vape activated charcoal quinoa hoodie try-hard artisan YOLO nisi dolor cardigan ugh. Pork belly salvia iPhone gluten-free retro you probably haven't heard of them seitan scenester whatever est chartreuse semiotics brooklyn hexagon gastropub."
            }, 
            {
                "id": 3,
                "name": "Fred",
                "uid": "00mKY5N1WBRO7mGcfpvFDzSGkJJ3",
                "complaint": "Lorem ipsum dolor amet copper mug labore neutra, artisan salvia squid green juice williamsburg literally ea typewriter small batch humblebrag seitan. Trust fund proident enim, activated charcoal pop-up chambray shabby chic officia woke. Church-key austin next level, scenester dreamcatcher succulents asymmetrical drinking vinegar. Keytar reprehenderit in hammock woke eu gochujang incididunt before they sold out ut chartreuse excepteur man bun locavore. Vape activated charcoal quinoa hoodie try-hard artisan YOLO nisi dolor cardigan ugh. Pork belly salvia iPhone gluten-free retro you probably haven't heard of them seitan scenester whatever est chartreuse semiotics brooklyn hexagon gastropub."
            }, 
            {
                "id": 4,
                "name": "Tara",
                "uid": "00mKY5N1WBRO7mGcfpvFDzSGkJJ3",
                "complaint": "Lorem ipsum dolor amet copper mug labore neutra, artisan salvia squid green juice williamsburg literally ea typewriter small batch humblebrag seitan. Trust fund proident enim, activated charcoal pop-up chambray shabby chic officia woke. Church-key austin next level, scenester dreamcatcher succulents asymmetrical drinking vinegar. Keytar reprehenderit in hammock woke eu gochujang incididunt before they sold out ut chartreuse excepteur man bun locavore. Vape activated charcoal quinoa hoodie try-hard artisan YOLO nisi dolor cardigan ugh. Pork belly salvia iPhone gluten-free retro you probably haven't heard of them seitan scenester whatever est chartreuse semiotics brooklyn hexagon gastropub."
            }, 
            {
                "id": 5,
                "name": "Sarah",
                "uid": "00mKY5N1WBRO7mGcfpvFDzSGkJJ3",
                "complaint": "Lorem ipsum dolor amet copper mug labore neutra, artisan salvia squid green juice williamsburg literally ea typewriter small batch humblebrag seitan. Trust fund proident enim, activated charcoal pop-up chambray shabby chic officia woke. Church-key austin next level, scenester dreamcatcher succulents asymmetrical drinking vinegar. Keytar reprehenderit in hammock woke eu gochujang incididunt before they sold out ut chartreuse excepteur man bun locavore. Vape activated charcoal quinoa hoodie try-hard artisan YOLO nisi dolor cardigan ugh. Pork belly salvia iPhone gluten-free retro you probably haven't heard of them seitan scenester whatever est chartreuse semiotics brooklyn hexagon gastropub."
            }
        ]
    }

    render() {
        return (
            <>
                {/* <Router> */}
                    <Navigation />
                    <div className='Homepage Container'>
                    <Link to='/record-complaint'>
                        <button>
                            <MaterialIcon icon="phone" />
                            Call and Complain
                        </button>
                    </Link>
                        <h1>
                            Chart of Worst Companies Goes Here
                        </h1>
                        <div>
                            {this.state.cards.map((card, i) => {
                                return <ComplaintCard card={card} key={this.state.id}/> 
                            })}
                        </div>
                    </div>
                    {/* <Route exact path="/" component={Home} /> */}
                    {/* <Route path="/record-complaint" component={RecordForm} />
                </Router> */}
            </>
        )
    }
}