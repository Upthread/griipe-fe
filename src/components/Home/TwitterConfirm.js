import React, { Component } from "react";
import { BrowserRouter as Router, Link, Route, Switch } from "react-router-dom";
import { withRouter } from "react-router";
import MuiThemeProvider from "material-ui/styles/MuiThemeProvider";
// import AppBar from "material-ui/AppBar";
// import TextField from "material-ui/TextField";
import RaisedButton from "material-ui/RaisedButton";
import MaterialIcon, {colorPalette} from 'material-icons-react';
import * as firebase from "firebase";
// import axios from 'axios';
import { Spinner, Fade, NavLink } from 'reactstrap';
import CloudDoneIcon from '@material-ui/icons/CloudDone';
// import { AuthUserContext } from '../Session/session.js';
import { withAuthorization } from '../Session/session.js';
import Navigation from '../Navigation/navigation.js';


class TwitterConfirm extends Component {
  state = {
    isLoading: true,
  }

  componentDidMount() {
    setTimeout(() => this.setState({isLoading: false}), 1000);
  }




  render() {
    // console.log(this.data)
    const { values, handleChange } = this.props;
    if(this.state.isLoading===true) {
      return (
      <div className="recording-loader loader">
        <h1>GRIIPE</h1>
        <br />
        <Spinner style={{ width: '3rem', height: '3rem' }} />
      </div>)
    };
    // console.log(firebase.auth().currentUser.displayName,
    // firebase.auth().currentUser.email,
    // firebase.auth().currentUser.uid,
    // this.props.StoreName,
    // this.props.StoreAddress,
    // this.props.StorePhone,
    // this.props.StoreGoogleRating,
    // this.props.StoreWebsite,
    // this.props.confirmationTranscription,
    // 0)
    return (
      <MuiThemeProvider>
        <Navigation />
        <Fade in={this.state.fadeIn} tag="h5" className="mt-3 form-container" >
        <h1 className="form-container-header">Your Tweet Has Been Sent</h1>
          <div className="confirmation-container">
            <span className="confirmation-span">
            <CloudDoneIcon color='error'/>
            <p className="confirmation-input"><strong>Link:</strong> {this.props.StoreName}</p>
            </span>
          </div>
          <div className="confirmation-container">
            <span className="confirmation-span">
              <CloudDoneIcon color='error'/>            
              <p className="confirmation-input"><strong>Tweet:</strong> {this.props.StoreAddress}</p>
            </span>
          </div>
          <Link to="/home" >
            <RaisedButton
                label="Home Page"
                primary={true}
                style={styles.button}
                // onClick={this.tweetAndRoute}
            />
          </Link>
        </Fade>
      </MuiThemeProvider>
    );
  }
}

const styles = {
  button: {
    margin: 15
  }
};

const condition = authUser => !!authUser;

export default withAuthorization(condition)(TwitterConfirm);