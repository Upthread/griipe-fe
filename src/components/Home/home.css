

.worstReviewed {
    text-align: center;
    margin-bottom: 30px;
}

.complaintButton {
    position: relative;
    width: 300px;
    height: 40px;
    /* margin-bottom: 20px;
    margin-top: 20px; */
    border-radius: 5px;
    background-color: #DA2C38;
    color: white;
    border: 1px solid #FFD6D6;
    text-decoration: none;
    z-index: 5;
    /* top: -80px; */
}

h1 {
    font-family: Lato,sans-serif;
    font-weight: 700;
    color: #343434
}

.centered {
    margin: 1px;
    position: relative;
    width: 302px;
    z-index: 4;
    top: -60px;
    
}

.button-container {
    
    display: flex;
    justify-content: center;
    length: 40px;
}

/* background: linear-gradient(#ff471a, red); */


.Homepage {
    background-color: #FFF8E8;
}

.profileCard {
    box-sizing: border-box;
    width: 210px;
    height: 320px;
    border: 1px solid lightgrey;
    border-radius: 15px;
    display: flex;
    flex-direction: column;
    align-items: center;
    margin-top: 10px;
    margin-right: 20px;
    overflow: hidden;
}

.profileCard img {
    width: 208px;
    height: 150px;
    object-fit: cover;
    margin-top: 20px;
}

.profileCard p {
    font-size: 1rem;
    font-family: Lato,sans-serif; 
    font-weight: 700;
}

.CardImg {

    width: 100%;

}

.imgdiv {
    display: flex;
    justify-content: center;
    width: 100%;
    border-bottom: 1px solid lightgrey;
    margin-bottom: 10px;
    background: #DA2C38;
}

.EditBtn {
    background: #DA2C38;
    border: 1px solid #DA2C38;
    width: 80%;
    height: 35px;
    border-radius: 2px;
}

.EditBtn:hover {
    background: #c12631;
}

.HomeWrapper {
    display: flex;
    justify-content: center;
}


@media (max-width: 550px) {

    /* .centered {
        margin: 1px;
        position: relative;
        width: 302px;
        top: 10px;
    } */

    
}

@media (max-width: 770px) {

    .centered {
        margin: 1px;
        position: relative;
        width: 302px;
        top: 10px;
    }

    h1 {
        margin-top: 50px;
    }

}