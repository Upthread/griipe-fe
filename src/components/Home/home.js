import React, { Component } from "react";
import { Link } from "react-router-dom";
import "./home.css";
import axios from "axios";
import * as firebase from "firebase";

import { withAuthorization } from "../Session/session.js";
import Navigation from "../Navigation/navigation.js";

import ComplaintCard from "../Feeds/ComplaintCard.js";
import ComplaintCardNoAuth from "../Feeds/ComplaintCardNoAuth.js";
import Chart from "../Chart/Chart.js";
import { Spinner, Fade } from "reactstrap";

import MaterialIcon, { colorPalette } from "material-icons-react";

import { AuthUserContext } from "../Session/session.js";

const HomePage = () => (
  <AuthUserContext.Consumer>
    {authUser => (authUser ? <HomePageWithAuth /> : <HomePageNoAuth />)}
  </AuthUserContext.Consumer>
);

class HomePageWithAuth extends Component {
  state = {
    complaintFeed: [],
    loading: true
  };

  user = firebase.auth().currentUser;

  ProfilePush = () => {
    this.props.history.push(`/edit-profile`);
  };

  componentDidMount() {
    this.complaints();
    // console.log("is this working?");
  }

  user = firebase.auth().currentUser;

  ProfilePush = () => {
    this.props.history.push(`/edit-profile`);
  };

  StoreNamess = () => {
    return this.state.complaintFeed.map(item => {
      return item.StoreName;
    });
  };

  complaints = () => {
    axios
      .get("https://griipe.herokuapp.com/api/routes/posts")
      .then(response => {
        this.setState({ complaintFeed: response.data, loading: false });
      })
      .catch(error => {
        console.error(error);
      });
  };

  render() {
    return (
      <>
        {/* <Fade tag="h5" className="mt-3"> */}
        <Navigation />
        <div className="Homepage Container">
          <div class="button-container">
            <Link class="centered" to="/complaint-form">
              <button class="complaintButton">Leave A Review</button>
            </Link>
          </div>
          <h1 class="worstReviewed">Lowest Reviewed Companies</h1>
          <div class="HomeWrapper">
            <div>
              {this.state.complaintFeed.map((card, i) => {
                return <ComplaintCard key={i} card={card} />;
              })}
            </div>
            <div class="BarGraph">
              <Chart StoreArray={this.StoreNamess()} />
            </div>
          </div>
        </div>
      </>
    );
  }
}

class HomePageNoAuth extends Component {
  state = {
    complaintFeed: []
  };

  componentDidMount() {
    this.complaints();
    console.log("is this working?");
  }

  user = firebase.auth().currentUser;

  ProfilePush = () => {
    this.props.history.push(`/edit-profile`);
  };

  StoreNamess = () => {
    return this.state.complaintFeed.map(item => {
      return item.StoreName;
    });
  };

  complaints = () => {
    axios
      .get("https://griipe.herokuapp.com/api/routes/posts")
      .then(response => {
        this.setState({ complaintFeed: response.data });
      })
      .catch(error => {
        console.error(error);
      });
  };

  render() {
    console.log(this.StoreNamess());
    console.log("ssdfsdf");
    return (
      <>
        <Navigation />
        <div className="Homepage Container">
          <div class="button-container">
            <Link class="centered" to="/complaint-form">
              <button class="complaintButton">
                Leave A Review
                {/* <MaterialIcon icon="phone" /> */}
              </button>
            </Link>
          </div>
          <h1 class="worstReviewed">Lowest Reviewed Companies</h1>
          <div class="HomeWrapper">
            <div>
              {this.state.complaintFeed.map((card, i) => {
                return <ComplaintCardNoAuth key={i} card={card} />;
              })}
              {console.log(this.state.complaintFeed)}
            </div>
            <div class="BarGraph">
              <Chart StoreArray={this.StoreNamess()} />
            </div>
          </div>
        </div>
      </>
    );
  }
}

// const condition = authUser => !!authUser;

// export default withAuthorization(condition)(HomePage);
export default HomePage;
