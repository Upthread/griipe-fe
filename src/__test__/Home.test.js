import React from "react";
import test from "react-test-renderer";

import Home from "../components/Home/home"


describe("<Home />", () => {
  it("matches snapshot", () => {
    const tree = test.create(<Home />);

    expect(tree.toJSON()).toMatchSnapshot();
  });
});