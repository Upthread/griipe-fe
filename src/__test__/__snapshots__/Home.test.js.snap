// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> matches snapshot 1`] = `
Array [
  <div>
    <nav
      className="navbar navbar navbar-expand-md navbar-light bg-light"
    >
      <a
        className="navbar-brand"
        href="/"
      >
        <i>
          Griipe
        </i>
      </a>
      <button
        className="navbar-toggler"
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="collapse navbar-collapse"
        style={Object {}}
      >
        <ul
          className="ml-auto navbar-nav"
        />
      </div>
    </nav>
  </div>,
  <div
    className="recording-loader loader"
  >
    <h1>
      Griipe
    </h1>
    <br />
    <div
      className="spinner-border"
      role="status"
      style={
        Object {
          "height": "3rem",
          "width": "3rem",
        }
      }
    >
      <span
        className="sr-only"
      >
        Loading...
      </span>
    </div>
  </div>,
]
`;
